yamux/src/connection.rs

```diff
@@ -350,7 +353,7 @@ struct Active<T> {
     socket: Fuse<frame::Io<T>>,
     next_id: u32,
 
-    streams: IntMap<StreamId, Stream>,
+    streams: IntMap<StreamId, Arc<Mutex<stream::Shared>>>,
```

Comment:
I wonder if it would turn out cleaner if `Shared` would hold `Arc<Mutex>` internally? That would remove the boilerplate around locking and some noise from the type signatures.

Reply:
Would prefer to merge the fix and leave this for future work if that's ok with you, since it looks like a bit more involved of a change